tooling/gatsby-plugin/src/provider.js

```diff
@@ -0,0 +1,36 @@
+import * as React from "react"
+import {
+  ThemeProvider,
+  ColorModeProvider,
+  CSSReset,
+  GlobalStyle,
+  PortalManager,
+  extendTheme,
+} from "@chakra-ui/react"
+import userTheme from "./theme"
+
+const theme = extendTheme(userTheme)
```

Comment:
I also added this because https://chakra-ui.com/guides/integrations/with-gatsby#chakra-providers also shows this

Reply:
I'd recommend to move the `extendTheme` function call back into userland code.
There is a feature landing soon, which adds theme extensions and if a gatsby user wants to apply one of those, she would need to shadow this file.

Reply:
@TimKolberger Ok, good to know! What about the bug I linked that was fixed with using `extendTheme()`?

If we go the route to move this to userland we'll need to update the docs.

Reply:
We should update the docs, because this approach lets the user control how the theme is created.
And because it previously worked only with `extendTheme` in `theme.js` I see it as a documentation flaw.

If a user wants to create its own application theme and provide a theme extension (in the near future), only the file `theme.js` needs to be shadowed.

Reply:
Done!